tasks/manager/src/test/java/org/infinispan/tasks/DummyTaskEngine.java

```diff
@@ -45,6 +46,9 @@ public String getName() {
       switch (DummyTaskTypes.valueOf(taskName)) {
       case SUCCESSFUL_TASK:
          return (CompletableFuture<T>) CompletableFuture.completedFuture("result");
+      case PARAMETERIZED_TASK:
+         Map<String, ?> params = context.getParameters().get();
```

Comment:
what if is Optional.empty ? In any case, for me this is an anti-pattern.
We should not be using an Optional<Map> this way but an empty map IMO

Reply:
This is a test. Not bothering with making this perfect